### PR TITLE
feat(python): Aws Bedrock embeddings integration

### PR DESCRIPTION
**Pull request by AyushExel**
Supports amazon titan, cohere english & cohere multi-lingual base models.

### CODE REVIEW

**Comment by AyushExel** [comment]
Hi @pkundu25, this PR adds support for all bedrock text embedding functions, you can use this branch for your workflow while this gets merged in main and is released

**Review by changhiskhan — COMMENTED** [review]
minor nits. please fix the doctest

File: python/lancedb/embeddings/bedrock.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+            )
+
+            # format output based on provider
+            response_body = json.loads(response.get("body").read())
+            if provider == "cohere":
+                return response_body.get("embeddings")[0]
+            else:
+                # includes common provider == "amazon"
+                return response_body.get("embedding")
+        except Exception as e:
```
[the remark this answers is not in the document]
Comment (AyushExel) [reply python/lancedb/embeddings/bedrock.py:167]: updated the help text. The exception can occur due to 2 reasons:
- The wrong json string format which should be pretty evident by the error msg
- Invalid/expired credentials, checking that is not straightforward so we provide a help text

File: python/lancedb/embeddings/bedrock.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+        texts: list[str] or np.ndarray (of str)
+            The texts to embed
+
+        Returns
+        -------
+        list[list[float]]
+            The embeddings for the given texts
+        """
+        results = []
+        for text in texts:
```
[the remark this answers is not in the document]
Comment (AyushExel) [reply python/lancedb/embeddings/bedrock.py:119]: I only see a string at a time in API docs https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/bedrock-runtime/client/invoke_model.html

File: python/tests/test_embeddings_slow.py [hunk 3]
```diff
@@ -202,3 +202,38 @@ class TextModel(LanceModel):
...
+
+
+@pytest.mark.slow
+@pytest.mark.skipif(
+    not aws_setup(), reason="AWS credentials not set or libraries not installed"
+)
+def test_bedrock_embedding(tmp_path):
+    for name in [
+        "amazon.titan-embed-text-v1",
+        "cohere.embed-english-v3",
```
[the remark this answers is not in the document]
Comment (AyushExel) [reply python/tests/test_embeddings_slow.py:225]: Yeah but even if they turn out to be different, it's not something that we can resolve?

**Review by changhiskhan — APPROVED** [review]
try rebasing to fix the GHA failure?

**Pull request merged**